apiproduct-operations/README.md

```diff
@@ -0,0 +1,202 @@
+checks that the current executing operation - using the REST model, this is a
+verb+path combination - is included within one of the API Products that is
+authorized for that credential. If not, then the request is rejected.
+
+In our example, for a `GET /contracts` request, the credential must be authorized
+for the viewer or admin product. For a `POST /contracts` request, the credential
+must be authorized for either creator or admin.
+
+## Implementation in the API Proxy
+
+This sample uses a simple API Proxy to demonstrate this function. The proxy uses
+a
+[VerifyAPIKey](https://cloud.google.com/apigee/docs/api-platform/reference/policies/verify-api-key-policy)
+policy to identify the calling application and the API product associated with
+the proxy. All of the magic of authorizing the operation for the given
+credential is done _implicitly_ by the VerifyAPIKey policy. There's nothing else
+you need to do in the API Proxy to make that happen.
+
+For the purposes of demonstration, the proxy uses an
+[AssignMessage](https://cloud.google.com/apigee/docs/api-platform/reference/policies/assign-message-policy)
+policy to set a mock success response, which includes the API product name, and
+the operation that was authorized.
+
+There's also a conditional flow in the proxy that uses [OAuthV2/VeriryAccessToken](https://cloud.google.com/apigee/docs/api-platform/reference/policies/oauthv2-policy#verifyaccesstoken) in place of VerifyAPIKey.
+The behavior with regard to operations authorization checks is exactly the same.
+
+
+## Screencast
+
+[![Alt text](./img/screenshot-20230828-140220.png)](https://www.youtube.com/watch?v=ep7h_tGHtiw)
+
+## Prerequisites
+
+1. [Provision Apigee X](https://cloud.google.com/apigee/docs/api-platform/get-started/provisioning-intro)
+2. Configure [external access](https://cloud.google.com/apigee/docs/api-platform/get-started/configure-routing#external-access) for API traffic to your Apigee X instance
+3. Permissions to create and deploy proxies, to create products, and to register apps and developers in Apigee. Get these permissions via the Apigee orgadmin role, or the combination of two roles: API Admin and Developer Admin. ([more on Apigee-specific roles](https://cloud.google.com/apigee/docs/api-platform/system-administration/apigee-roles#apigee-specific-roles))
+4. Make sure the following tools are available in your terminal's $PATH (Cloud Shell has these preconfigured)
+    * [gcloud SDK](https://cloud.google.com/sdk/docs/install)
+    * unzip
+    * curl
+    * jq
+    * npm
+
+## (QuickStart) Setup using CloudShell
+
+Use the following GCP CloudShell tutorial, and follow the instructions.
+
+[![Open in Cloud Shell](https://gstatic.com/cloudssh/images/open-btn.png)](https://ssh.cloud.google.com/cloudshell/open?cloudshell_git_repo=https://github.com/DinoChiesa/apigee-samples&cloudshell_git_branch=main&cloudshell_workspace=.&cloudshell_tutorial=apiproduct-operations/docs/cloudshell-tutorial.md)
```

Comment:
the URL is pointing to your personal repo

Reply:
fixed.